pkg/runtime/wait.go

```diff
@@ -69,3 +75,108 @@ func checkIfOutboundReady(client *http.Client, outboundReadyHealthURL string) er
 
 	return nil
 }
+
+func (a *DaprRuntime) blockUntilAppPortOpen() {
+	if a.runtimeConfig.ApplicationPort <= 0 {
+		return
+	}
+
+	log.Infof("application protocol: %s. waiting on port %v.  This will block until the app is listening on that port.", string(a.runtimeConfig.ApplicationProtocol), a.runtimeConfig.ApplicationPort)
+
+	for {
+		conn, _ := net.DialTimeout("tcp", net.JoinHostPort("localhost", fmt.Sprintf("%v", a.runtimeConfig.ApplicationPort)), time.Millisecond*500)
+		if conn != nil {
+			conn.Close()
+			break
+		}
+		// prevents overwhelming the OS with open connections
+		time.Sleep(time.Millisecond * 50)
+	}
+
+	log.Infof("application discovered on port %v", a.runtimeConfig.ApplicationPort)
+}
+
+func (a *DaprRuntime) blockUntilAppIsReady() {
+	if a.runtimeConfig.Mode == modes.KubernetesMode && len(a.runtimeConfig.WaitingContainers) > 0 {
+		log.Infof("going to check containers ready : namespace = %s, podName = %s, containers = %v", a.namespace, a.podName, a.runtimeConfig.WaitingContainers)
+	out:
+		for {
+			// check readiness from k8s env
+			resp, err := a.operatorClient.GetContainersStatus(context.Background(), &operatorv1pb.GetContainersStatusRequest{
+				Namespace: a.namespace,
+				Name:      a.podName,
+			})
+			if err != nil {
+				log.Errorf("get containers status error : namespace = %s, podName = %s, err = %s", a.namespace, a.podName, err)
+				return
+			}
+
+			if contains(a.runtimeConfig.WaitingContainers, "all") {
+				// all containers in pod should be ready
+				for name, ready := range resp.Statuses {
+					if name == "daprd" {
+						continue
+					}
+					if !ready {
+						log.Infof("container[%s] not ready, continue waiting", name)
+						time.Sleep(time.Millisecond * 1000)
+						continue out
+					}
+				}
+			} else {
+				// all specified containers should be ready
+				for _, name := range a.runtimeConfig.WaitingContainers {
+					ready, ok := resp.Statuses[name]
+					if !ok {
+						log.Infof("container[%s] not found, continue waiting", name)
+						time.Sleep(time.Millisecond * 1000)
+						continue out
+					}
+					if !ready {
+						log.Infof("container[%s] not ready, continue waiting", name)
+						time.Sleep(time.Millisecond * 1000)
+						continue out
+					}
+				}
+			}
+
+			log.Infof("all containers are ready")
+			break
+		}
+	}
+
+	if a.runtimeConfig.WaitingProbe != "" {
```

Comment:
Doesn't K8s already have readiness check? I see that this is helpful for Standalone mode but this is too specific - it is on HTTP only (does not support gRPC) and content-type is forced to JSON. Can the waitingProbe (maybe readinessCheck is a better name here) be configured in the Configuration CRD?

Again, this check only adds value to standalone mode.